src/__tests__/__helpers__/mockClerkServer.ts

```diff
@@ -0,0 +1,87 @@
+import * as clerk from "@clerk/nextjs/server";
+export type CurrentUser = NonNullable<
+  Awaited<ReturnType<typeof clerk.currentUser>>
+>;
+
+/**
+ * Sets the return value of `currentUser()` on the server
+ */
+export function setCurrentUser(user: CurrentUser | null) {
+  jest.spyOn(clerk, "currentUser").mockResolvedValue(user);
+}
+
+/**
+ * Installs a partial mock of `clerkClient()` for use on the server
+ * to make requests to Clerk's API
+ */
+export function installMockClerkClient() {
+  const mockClerkClient = {
+    users: {
+      updateUserMetadata: () => Promise.resolve(mockCurrentUser),
+    },
+  } as unknown as ReturnType<typeof clerk.clerkClient>;
+  jest.spyOn(clerk, "clerkClient").mockReturnValue(mockClerkClient);
+  return mockClerkClient;
+}
```

Comment:
Makes it easier to mock the Clerk server module in tests by providing some utilities to set test fixtures.